sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala

```diff
@@ -246,6 +246,34 @@ class HiveProviderImpl extends HiveProvider {
             if (extendedBool && hasBooleans) {
               willNotWorkOnGpu("extended boolean parsing is not supported")
             }
+
+            lazy val hasFloats = wrapped.output.exists { att =>
+              TrampolineUtil.dataTypeExistsRecursively(att.dataType, dt => dt == FloatType)
+            }
+
+            if (!conf.shouldHiveReadFloats && hasFloats) {
+              willNotWorkOnGpu("reading of floats has been disabled set " +
+                  s"${RapidsConf.ENABLE_READ_HIVE_FLOATS} to true to enable this.")
+            }
+
+            lazy val hasDoubles = wrapped.output.exists { att =>
+              TrampolineUtil.dataTypeExistsRecursively(att.dataType, dt => dt == DoubleType)
+            }
+
+            if (!conf.shouldHiveReadDoubles && hasDoubles) {
+              willNotWorkOnGpu("reading of doubles has been disabled set " +
+                  s"${RapidsConf.ENABLE_READ_HIVE_DOUBLES} to true to enable this.")
+            }
+
+            lazy val hasDecimals = wrapped.output.exists { att =>
+              TrampolineUtil.dataTypeExistsRecursively(att.dataType,
+                dt => dt.isInstanceOf[DecimalType])
+            }
+
+            if (!conf.shouldHiveReadDecimals && hasDecimals) {
```

Comment:
My gut reaction to this was to consider a switch for `TIMESTAMP`. The only known issue is in the case of `timestamp.formats`, which is its own switch.

Reply:
I will add it.